R/example.R

```diff
@@ -195,4 +195,57 @@ example_large_sized_commits <- function(folder_path="/tmp", folder_name) {
 }
 
 
+#' Example fake Mbox data with all parameters
+#'
+#' Creating a fake mbox with all parameters filled in, to ensure the function generates mbox properly
+#'
+#' @param folder_path Default folder path set to "/tmp"
+#' @param folder_name Name of folder where .mbox file will be stored
+#' @return Folder path of .mbox sample file that was created
+#' @export
+example_mbox_normal <- function(folder_path = "/tmp", folder_name) {
+
+  # Define sample data for create_fake_mbox function that can easily be altered for testing
+  mlist <- "test-list"
+  reply_from_author <- "John Doe"
+  reply_from_email <- "johndoe@example.com"
+  reply_to_author <- "Janette Doe"
+  reply_to_email <- "janedoe@example.com"
+  reply_cc_author <- "Smithsonian Doe"
+  reply_cc_email <- "smith_doe@example.com"
+  reply_datetime <- "2023-01-15T08:30:00"
+  timezone <- "EST"
+  reply_subject <- "Test Email Subject"
+  reply_body <- "This is the body of the test email."
+
+
+  # Create fake mbox using create_fake_mbox function
+  mbox_content <- create_fake_mbox(
+    mlist = mlist,
+    reply_from_author = reply_from_author,
+    reply_from_email = reply_from_email,
+    reply_to_author = reply_to_author,
+    reply_to_email = reply_to_email,
+    reply_cc_author = reply_cc_author,
+    reply_cc_email = reply_cc_email,
+    reply_datetime = reply_datetime,
+    timezone = timezone,
+    reply_subject = reply_subject,
+    reply_body = reply_body
+  )
+
+
+  # Create a unique filename for the mbox file
+  git_init(folder_path)
+  mbox_filepath <- file.path(folder_path, "sample.mbox")
+
+  # Write the mbox content to a file
+  writeLines(mbox_content, mbox_filepath)
```

Comment:
The mail fake data should have a function that creates the .mbox file, not the example. The features are "leaking" out of the fake data generator. Think about it, the git_* functions know how to commit, initialize repos, stage files, etc. the io_ knows how to create files with content. For consistency, the mbox function should know how to make mbox files. 

I will elaborate on this on the PR main comment than this inline.